src/lib.rs

```diff
@@ -1528,6 +1528,60 @@ fn render_table_cell<T: Write, D: TextDecorator>(
     })
 }
 
+pub mod config {
+    //! Configure the HTML to text translation using the `Config` type, which can be
+    //! constructed using one of the functions in this module.
+
+    use crate::render::text_renderer::{
+        PlainDecorator, RichDecorator, TaggedLine, TextDecorator
+    };
+    use super::parse;
+
+    /// Configure the HTML processing.
+    pub struct Config<D: TextDecorator> {
+        decorator: D,
+    }
+
+    impl<D: TextDecorator> Config<D> {
+        /// Reads HTML from `input`, and returns a `String` with text wrapped to
+        /// `width` columns.
+        pub fn string_from_read<R: std::io::Read>(self, input: R, width: usize) -> String {
```

Comment:
These methods are destructive (they can only be used once), which I guess is probably ok.  Otherwise either the decorators would need to be `Clone`, or it would have to take a "Decorator maker" instead.